shoebot/grammar/nodebox.py

```diff
@@ -624,16 +622,36 @@ def background(self, *args):
 
     # Text
 
-    def font(self, fontpath=None, fontsize=None):
+    def font(self, fontpath=None, fontsize=None, vars=None, *args, **kwargs):
         """Set the font to be used with new text instances.
 
-        :param fontpath: path to truetype or opentype font.
-        :param fontsize: size of font
+        :param fontpath: font name (can include styles like "Bold")
+        :param fontsize: font size
+        :param vars: font variant values, as a dict of axis/value pairs (variable fonts only)
+        :param var_XXXX: set variant value (variable fonts only)
 
         :return: current current fontpath (if fontpath param not set)
         Accepts TrueType and OpenType files. Depends on FreeType being
         installed."""
         if fontpath is not None:
+            # do we have variants set?
+            if not vars:
+                variants = {}
```

Comment:
Consider making variants a list of strings, like `["wdth=50", "wght=600"]` then the code below will get simpler...

The first loop can then be a list comprehension:

```python
variants = [f"{arg.replace('var_', '')={value}}" for arg, value in kwargs.items() if arg.startswith("var_")
```
# ^ Black does something interesting to the format of this..

Then the logic at the end can be something along the lines of 

```python
fontpath += " @" + variants.join(",")
```

Reply:
great suggestion, pushed in d5f4b15056c93fc2de5906a37bbe3a49c0dda030